### PR TITLE
Corpus: Language Detection

### PR DESCRIPTION
**Pull request by djukicn**
##### Issue
Implements #582. 

##### Description of changes
- Added the language detection functionality to the Corpus widget.

##### Includes
- [X] Code changes
- [X] Tests
- [ ] Documentation

### CODE REVIEW

**Comment by ajdapretnar** [comment]
This needs a rebase (it's a very simple one, just include both tests).

I am getting an error with the following workflow:
Twitter (retrieve any 100 tweets) - Corpus (detect languages)

Error log:
```
Traceback (most recent call last):
  File "/Users/ajda/orange/orange-widget-base/orangewidget/gui.py", line 1949, in __call__
    self.func(**kwds)
  File "/Users/ajda/orange/orange3-text/orangecontrib/text/widgets/owcorpus.py", line 320, in handle_languages
    self.corpus.detect_languages()
  File "/Users/ajda/orange/orange3-text/orangecontrib/text/corpus.py", line 240, in detect_languages
    self.languages = [model.predict(t)[0][0].replace('__label__', '')
  File "/Users/ajda/orange/orange3-text/orangecontrib/text/corpus.py", line 240, in <listcomp>
    self.languages = [model.predict(t)[0][0].replace('__label__', '')
  File "/Users/ajda/opt/miniconda3/envs/o3/lib/python3.8/site-packages/fasttext/FastText.py", line 225, in predict
    text = check(text)
  File "/Users/ajda/opt/miniconda3/envs/o3/lib/python3.8/site-packages/fasttext/FastText.py", line 212, in check
    raise ValueError(
ValueError: predict processes one line at a time (remove '\n')
```

File: orangecontrib/text/widgets/owcorpus.py [hunk 1]
```diff
@@ -303,6 +312,60 @@ def describe(features):
...
+            domain = self.corpus.domain
+            if self.detect_languages:
+                if self.corpus.languages is None:
+                    self.corpus.detect_languages()
+
+                curr_attributes = list(domain.attributes)
+                curr_class_var = [domain.class_var] if domain.class_var else []
+                curr_metas = list(domain.metas)
+                curr_variables = curr_attributes + curr_class_var + curr_metas
+                curr_names = [var.name for var in curr_variables]
```
Comment (ajdapretnar) [inline orangecontrib/text/widgets/owcorpus.py:326]: I believe you can make this shorter with:
`[var.name for var in domain.variables]`
without any of the calls above.

File: orangecontrib/text/widgets/owcorpus.py [hunk 2]
```diff
@@ -303,6 +312,60 @@ def describe(features):
...
+                new_name = get_unique_names(curr_names, "Language")
+
+                variable_attrs = {'language-feature': True}
+                new_variable = StringVariable(new_name)
+                new_variable.attributes.update(variable_attrs)
+                new_domain = Domain(
+                    attributes=list(domain.attributes),
```
Comment (ajdapretnar) [inline orangecontrib/text/widgets/owcorpus.py:333]: Are you sure you need to cast this as a list?

File: orangecontrib/text/widgets/owcorpus.py (hunk 1 above)
Comment (djukicn) [reply orangecontrib/text/widgets/owcorpus.py:326]: This doesn't seem to include names of metas.

Comment (ajdapretnar) [reply orangecontrib/text/widgets/owcorpus.py:326]: You're right. I hate this. But keeping with a one-lines, this should work:
`[var.name for var in corpus.domain.variables + corpus.domain.metas]`

No need to fix it now, I just think it is easier to read and keeps the code base concise. :)

**Pull request merged**